core/net/ipv6/websocket-http-client.c

```diff
@@ -0,0 +1,337 @@
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#include "websocket-http-client.h"
+#include "net/ip/uiplib.h"
+#include "net/ip/resolv.h"
+
+#include "ip64-addr.h"
+
+#include <stdio.h>
+#include <string.h>
+
+#define DEBUG DEBUG_NONE
+#include "net/ip/uip-debug.h"
+
+enum {
+  STATE_WAITING_FOR_HEADER,
+  STATE_WAITING_FOR_CONNECTED,
+  STATE_STEADY_STATE,
+};
+/*---------------------------------------------------------------------------*/
+static void
+send_get(struct websocket_http_client_state *s)
+{
+  struct tcp_socket *tcps;
+
+  tcps = &s->s;
+  tcp_socket_send_str(tcps, "GET ");
+  tcp_socket_send_str(tcps, s->file);
+  tcp_socket_send_str(tcps, " HTTP/1.1\r\n");
+  tcp_socket_send_str(tcps, "Host: ");
+  tcp_socket_send_str(tcps, s->host);
+  tcp_socket_send_str(tcps, "\r\n");
+  if(strlen(s->header) > 0) {
+    tcp_socket_send_str(tcps, s->header);
+  }
+  tcp_socket_send_str(tcps,
+                      "Connection: Upgrade\r\n"
+                      "Upgrade: websocket\r\n"
+                      "Sec-WebSocket-Key: x3JJHMbDL1EzLkh9GBhXDw==\r\n"
+                      "Sec-WebSocket-Version: 13\r\n"
+                      "Sec-WebSocket-Protocol:");
+  tcp_socket_send_str(tcps, s->subprotocol);
+  tcp_socket_send_str(tcps, "\r\n");
+  tcp_socket_send_str(tcps, "\r\n");
+  PRINTF("websocket-http-client: send_get(): output buffer left %d\n", tcp_socket_max_sendlen(tcps));
+}
+/*---------------------------------------------------------------------------*/
+static void
+send_connect(struct websocket_http_client_state *s)
+{
+  struct tcp_socket *tcps;
+  char buf[20];
+
+  tcps = &s->s;
+  tcp_socket_send_str(tcps, "CONNECT ");
+  tcp_socket_send_str(tcps, s->host);
+  tcp_socket_send_str(tcps, ":");
+  sprintf(buf, "%d", s->port);
+  tcp_socket_send_str(tcps, buf);
+  tcp_socket_send_str(tcps, " HTTP/1.1\r\n");
+  tcp_socket_send_str(tcps, "Host: ");
+  tcp_socket_send_str(tcps, s->host);
+  tcp_socket_send_str(tcps, "\r\n");
+  tcp_socket_send_str(tcps, "Proxy-Connection: Keep-Alive\r\n\r\n");
+}
+/*---------------------------------------------------------------------------*/
+static void
+event(struct tcp_socket *tcps, void *ptr,
+      tcp_socket_event_t e)
+{
+  struct websocket_http_client_state *s = ptr;
+
+  if(e == TCP_SOCKET_CONNECTED) {
+    if(s->proxy_port != 0) {
+      send_connect(s);
+    } else {
+      send_get(s);
+    }
+  } else if(e == TCP_SOCKET_CLOSED) {
+    websocket_http_client_closed(s);
+  } else if(e == TCP_SOCKET_TIMEDOUT) {
+    websocket_http_client_timedout(s);
+  } else if(e == TCP_SOCKET_ABORTED) {
+    websocket_http_client_aborted(s);
+  } else if(e == TCP_SOCKET_DATA_SENT) {
```

Comment:
This check is unnecessary, for code readiness perhaps  adding just as a comment would be just fine